src/codegen/mod.rs

```diff
@@ -4381,7 +4381,7 @@ pub mod utils {
             }
         } else {
             quote! {
-                use objc;
+                use objc::{self, msg_send, sel, sel_impl, class};
```

Comment:
Are we worried about namespace pollution in the generated code with generic names like `class` and `self` ? This may be of no importance – I honestly do not have a good handle on the implications of bringing in these names unqualified.

Reply:
I will admit that I don't know if `self` is needed here. It was there before so I left it in now that you question it, it may not be needed. The (unfortunate) generic `class` however is needed for the non-`objc_extern_crate` scenario.

Before rust edition 2018 (if I recall the history), things like:
```rust
#[macro_use] extern crate objc;
```
were required for all external crates and imported all macros from that crate via the `#[macro_use]` annotation. With rust 2018, one doesn't need `extern crate foo` for crate foo on `src/lib.rs` or `src/main.rs` but as a result one cannot just blanket import all the macros (this is probably for the best).

One unfortunate part of this is that macros use macros but the macros don't import the macros from the crates they use that I know of and not from the crates they live in (for the objc crate anyway).

Anyway, I bring this up because I find the `#[macro_use]` annotation nice but only because it works now. If I'm being honest this crate should probably deprecate it but argue that it's outside the scope of this PR.

Reply:
I agree that deprecation is out of scope for this PR. I am pretty weak on the edition differences and macro scoping rules, but after your explanation I think the namespace collision problem is not worse than it was. Thanks.